GameServerLib/ObjectManager.cs

```diff
@@ -135,9 +135,15 @@ public void Update(float diff)
                 if (ai != null)
                 {
                     var tempBuffs = ai.GetBuffs();
-                    foreach (var buff in tempBuffs.Values)
-                    {                       
-                        buff.Update(diff);
+                    for (int i = 0; i < tempBuffs.Count; i++)
+                    {
+                        if (tempBuffs[i].Elapsed())
+                        {
+                            ai.RemoveBuff(tempBuffs[i]);
+                            i--;
+                            continue;
+                        }
+                        tempBuffs[i].Update(diff);
```

Comment:
Isn't the current method more efficient? It only uses one for loop compared to your suggestion of two. I'm not familiar with how efficient the List.ForEach is, so I'm just verifying.

Reply:
Please fix this to start from the end (as we discussed) to avoid the changing of `i` inside the loop. My suggested solution I commented here is completely wrong.